pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/impl/TopicTransactionBuffer.java

```diff
@@ -43,6 +42,12 @@
 
     private final PersistentTopic topic;
 
+    private volatile PositionImpl maxReadPosition = PositionImpl.latest;
+
+    private final LinkedMap<TxnID, PositionImpl> ongoingTxns = new LinkedMap<>();
+
+    private final LinkedMap<TxnID, PositionImpl> aborts = new LinkedMap<>();
```

Comment:
Why this also need LinkedMap ? Please give comment for these two maps especially the value of the map, I think the first map is the min message position of the transaction and the second map is the max message position of the transaction?

Reply:
When we handle lowWaterMark, we should get the first entry of the map to compare the txnID with lowWaterMark. And ongoingTxns is ordered, so we should get the first key position become the maxReadPosition.